### ISSUE TITLE
Create Custom Hooks with React Query for Handling Authentication Requests

### ISSUE THREAD

**Issue by AdityaGhamat** [LakshmiSowmya04/GradLink#14]
We need to create custom hooks using React Query for handling authentication-related requests in our application. This includes useSignup for user registration, useLogin for user login, and useOtpVerification for verifying OTPs. Each hook should manage loading and error states, promoting code reusability and cleaner components.
Pros - 
1.Minimum use of useEffect, so performance can be improved so much.
2.Less code and modular code

**Comment by AdityaGhamat** [comment]
@LakshmiSowmya04 Its really essential issue and i wish to solve it.

**Comment by AdityaGhamat** [comment]
@LakshmiSowmya04 can you tag it in hacktoberfest?

**Comment by nikkuv** [comment]
@AdityaGhamat, yes, the hook will support requests coming from the server. I'm done with the task but wasn't able to push to the same repo. Will you make me a contributor in this repo, or do I have to fork the repo to push my changes?

**Comment by LakshmiSowmya04** [comment]
> @AdityaGhamat, yes, the hook will support requests coming from the server. I'm done with the task but wasn't able to push to the same repo. Will you make me a contributor in this repo, or do I have to fork the repo to push my changes?

U have to fork nd then raise pr
